readthedocs/proxito/middleware.py

```diff
@@ -176,6 +176,66 @@ def add_proxito_headers(self, request, response):
         else:
             response['X-RTD-Version-Method'] = 'path'
 
+    def add_user_headers(self, request, response):
+        """
+        Set specific HTTP headers requested by the user.
+
+        The headers added come from ``projects.models.HTTPHeader`` associated
+        with the ``Domain`` object.
+        """
+        if hasattr(request, 'domain'):
+            response_headers = [header.lower() for header in response]
+            for http_header in request.domain.http_headers.all():
+                if http_header.name.lower() in response_headers:
+                    log.error(
+                        'Overriding an existing response HTTP header. header=%s domain=%s',
+                        http_header.name,
+                        request.domain.domain,
+                    )
+                log.info(
+                    'Adding custtom response HTTP header. header=%s domain=%s',
+                    http_header.name,
+                    request.domain.domain,
+                )
+                response[http_header.name] = http_header.value
```

Comment:
Let's just put a comment here that the headers are limited by the validator/choices on the `HTTPHeader` model. Setting arbitrary HTTP headers is potentially dangerous (I realize this is staff-only but things change over time).